server/tests/test_notebook_view.py

```diff
@@ -149,6 +149,36 @@ def test_tryit_view(client, fake_user, logged_in, iomd):
         assert len(response.redirect_chain) == 0
 
 
+@pytest.mark.parametrize("logged_in", [True, False])
+@pytest.mark.parametrize("request_method", ["POST", "GET"])
+def test_tryit_view_file_uploads(client, fake_user, logged_in, request_method):
+    import tempfile
+    import mimetypes
+
+    # we should see the same script blocks
+    # for both logged_in and both request_method
+
+    with tempfile.NamedTemporaryFile(mode="w+") as f:
```

Comment:
There is no need to create an actual temporary file here. You can just upload some arbitrary content (e.g. a string).